File: rpg-frontend/src/types/enemy.ts
export interface EnemyOut {
  id: number;
  name: string;
  description: string | null;
  image_path: string | null;
  level: number;
  hp: number;
  attack: number;
  defense: number;
  speed: number;
  crit_chance: number;
  crit_damage: number;
  reward_xp: number;
  reward_coins: number;
  is_available_to_user: boolean;
}

File: rpg-backend/app/schemas/enemy.py
from pydantic import BaseModel


class EnemyOut(BaseModel):
    id: int
    name: str
    description: str | None
    image_path: str | None
    level: int
    hp: int
    attack: int
    defense: int
    speed: int
    crit_chance: float
    crit_damage: float
    reward_xp: int
    reward_coins: int
    is_available_to_user: bool
